internal/pkg/template/workload_test.go

```diff
@@ -479,3 +471,36 @@ func TestRollingUpdateRollbackConfig_TruncateAlarmName(t *testing.T) {
 		})
 	}
 }
+
+func TestApplicationLoadBalancer_Aliases(t *testing.T) {
+	tests := map[string]struct {
+		opts     ALBListener
+		expected []string
+	}{
+		"LBWS with multiple listener rules having multiple aliases each": {
```

Comment:
Do we need a test case that says `"LBWS with no aliases"`

Reply:
The real motto of this function to find out unique aliases across all the listener rules(ALB) or listeners(NLB) and return those aliases. That is why I didn't find a need for the test with no Aliases as it returns empty []string. But I have added that test for safety.